### PR TITLE
Squash game

### PR DESCRIPTION
**Pull request by nagado**
Little things fixed

### CODE REVIEW

**Comment by nagado** [comment]
Hey, I am ready to merge...?

File: sfml/squash_game/Ball.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+#include "TextException.h"
+#include "Utils.h"
+#include <math.h>
+#include "Ball.h"
+
+
+void Ball::move()
+{
+  if (angle > 2 * Utils::Pi || angle < 0 * Utils::Pi)
+    throw TextException("Unacceptable angle\n");
```
Comment (grok45) [inline sfml/squash_game/Ball.cpp]: Better to have exception class making some sense.

File: sfml/squash_game/Ball.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+void Ball::updatePosition(const std::vector<std::vector<int>> thing_borders, const int thing_speed, const double thing_angle)
+{
+  if (thing_borders.size() != 0) 
+    recalculateAngle(thing_borders, thing_speed, thing_angle);
+  move();
+}
+
+void Ball::draw(sf::RenderTarget& target, sf::RenderStates states) const
+{
+  int f = 1 - radius;
```
Comment (grok45) [inline sfml/squash_game/Ball.cpp:33]: Too long paragraph

File: sfml/squash_game/Ball.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class Ball : public Thing
+{
+  double x;
+  double y;
+  int radius;
+
+public:
+  Ball(int xx, int yy, int rad, double ang, double spe) : Thing(true, true, ang, spe), radius(rad) { setPosition(xx, yy); x = xx; y = yy; }
```
Comment (grok45) [inline sfml/squash_game/Ball.h]: Too long line

File: sfml/squash_game/Game.cpp [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+#include "Ball.h"
+#include "Racquet.h"
+#include <memory>
+
+
+Game::Game(sf::RenderWindow& window) 
+{
+  using namespace Utils;
+
+  objects.push_back(std::unique_ptr<Ball> (new Ball(int(window_x / 2 - ball_r), ball_start_height, ball_r, ball_angle, ball_speed)));
```
Comment (grok45) [inline sfml/squash_game/Game.cpp]: I would initialize those objects from an array.

File: sfml/squash_game/Game.h [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#ifndef GAME_H
```
Comment (grok45) [inline sfml/squash_game/Game.h:1]: usually better make a name starting with  __
like **GAME_H**

File: sfml/squash_game/Racquet.cpp [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+std::vector<std::vector<int>> Racquet::getBorders() const
+{
+  std::vector<std::vector<int>> borders {{int(getPosition().x - length / 2), int(getPosition().y - height / 2)}, 
+                                         {int(getPosition().x + length / 2), int(getPosition().y + height / 2)}};
+
+  return borders;
+}
+
+void Racquet::draw(sf::RenderTarget& target, sf::RenderStates states) const
```
Comment (grok45) [inline sfml/squash_game/Racquet.cpp:65]: Why this isn't a const reference?

File: sfml/squash_game/Thing.h [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  bool movable;
+  bool needs_updating;
+  double speed;
+  double angle;
+
+  Thing(bool mov, bool upd) : movable(mov), needs_updating(upd) {}
+  Thing(bool mov, bool upd, double ang, double spe) : movable(mov), needs_updating(upd), speed(spe), angle(ang) {}
+
+public:
+  virtual void changePosition(int xx, int yy) { setPosition(xx, yy); };
```
Comment (grok45) [inline sfml/squash_game/Thing.h:21]: It's a really long peace of code without empty lines...

File: sfml/squash_game/Thing.h [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  unsigned getPointCount() const override { return unsigned(1); }//assert on a call
+  sf::Vector2f getPoint(unsigned index) const override{return sf::Vector2f(1, 2); }//assert on a call
+  void setAngle(double a) { angle = a; }
+  double getAngle() const { return angle; }
+  void setSpeed(double s) { speed = s; }
+  double getSpeed() const { return speed; }
+  virtual void draw(sf::RenderTarget& target, sf::RenderStates states) const = 0;
+  virtual std::vector<std::vector<int>> getBorders() const = 0;//maybe return vector of Vertices
+
+  virtual ~Thing() {}
```
Comment (grok45) [inline sfml/squash_game/Thing.h]: Destructor better be right after constructors

File: sfml/squash_game/Utils.h [hunk 9]
```diff
@@ -0,0 +1,32 @@
+#ifndef UTILS_H
+#define UTILS_H
+
+
+namespace Utils
+{
+  extern const double Pi;
```
Comment (grok45) [inline sfml/squash_game/Utils.h]: Better replace all the "extern" with "const"

File: sfml/squash_game/ubuntu-font-family-0.83/CONTRIBUTING.txt [hunk 10]
```diff
@@ -0,0 +1,21 @@
+The Ubuntu Font Family is very long-term endeavour, and the first time
```
Comment (grok45) [inline sfml/squash_game/ubuntu-font-family-0.83/CONTRIBUTING.txt:1]: Nice file!

**Comment by grok45** [comment]
My comments are in the pull request

On Thu, Jan 21, 2016 at 12:39 PM, Elizaveta Kishchukova <
notifications@github.com> wrote:

> Hey, I am ready to merge...?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/nagado/educ/pull/7#issuecomment-173648986.

## 

Regards,
Dmitrii Kishchukov

File: sfml/squash_game/Game.cpp (hunk 4 above)
Comment (nagado) [inline sfml/squash_game/Game.cpp]: I am not actually sure what exactly do you mean, there could be different arrays for different solutions

File: sfml/squash_game/Game.h (hunk 5 above)
Comment (nagado) [inline sfml/squash_game/Game.h:1]: your example doesn't start with __. 
Also, from what I've heard, "According to the C++ Standard, identifiers starting with one underscore are reserved for libraries. Identifiers starting with two underscores are reserved for compiler vendors."(http://stackoverflow.com/questions/224397/why-do-people-use-double-underscore-so-much-in-c)

File: sfml/squash_game/Racquet.cpp (hunk 6 above)
Comment (nagado) [inline sfml/squash_game/Racquet.cpp:65]: 'cuz it's one I am overriding. I can not change this function's header unless I want it to override the SFML draw() function

**Pull request closed without merging**